src/index/IndexImpl.Text.cpp

```diff
@@ -1493,20 +1475,20 @@ void IndexImpl::getECListForWordsAndSingleSub(
     }
     // Test if each context is fitting.
     size_t currentContextFrom = 0;
-    TextRecordIndex currentContext = cids[0];
+    TextRecordIndex currentContext = wep.cids[0];
     bool matched = false;
-    for (size_t i = 0; i < cids.size(); ++i) {
-      if (cids[i] != currentContext) {
+    for (size_t i = 0; i < wep.cids.size(); ++i) {
+      if (wep.cids[i] != currentContext) {
         if (matched) {
-          FTSAlgorithms::appendCrossProduct(
-              cids, eids, scores, currentContextFrom, i, subEs, nonAggRes);
+          FTSAlgorithms::appendCrossProduct(wep, currentContextFrom, i, subEs,
+                                            nonAggRes);
         }
         matched = false;
-        currentContext = cids[i];
+        currentContext = wep.cids[i];
         currentContextFrom = i;
       }
       if (!matched) {
-        matched = (subEs.count(eids[i]) > 0);
+        matched = (subEs.count(wep.eids[i]) > 0);
```

Comment:
```suggestion
        matched = subEs.contains(wep.eids[i]);
```
In the meantime we have `contains` for HashMaps, so we can use some modernization:)